waiter/src/waiter/service_description.clj

```diff
@@ -340,7 +362,20 @@
        :service-id service-id}))
 
   (validate [_ service-description args-map]
-    (validate-schema service-description (merge-with set/union args-map {:valid-cmd-types #{"shell"}}))))
+    (->> (merge-with set/union args-map {:valid-cmd-types #{"shell"}})
+         (validate-schema service-description upper-limits-schema))))
+
+(defn create-default-service-description-builder
+  "Returns a new DefaultServiceDescriptionBuilder which uses the specified resource limits."
+  [{:keys [constraints]}]
+  (let [upper-limits (->> constraints
```

Comment:
Would it be better to pass the full constraints to the builder, instead of a projection of some of the constraints?  This logic could belong inside the builder.

Reply:
We need to build it here as `DefaultServiceDescriptionBuilder` is a record. We can create the new constraints when we support them and pass them along.